utils.js

```diff
@@ -234,7 +236,9 @@ export function getLicenses(pkg, format = "xml") {
         }
         return licenseContent;
       })
-      .map((l) => ({ license: l }));
+      .map((l) =>
+        l.expression ? { expression: l.expression } : { license: l }
```

Comment:
I think there can be only one expression, so possible we are returning multiple.

Reply:
The `licenses` list supports multiple schemas in the array. I'm not certain if the spec requires all data to be the same schema in the list. [Spec](https://cyclonedx.org/docs/1.5/json/#components_items_licenses_oneOf_i1_items_i0_expression)

Reply:
That appears to be the difference between CycloneDX 1.4 and 1.5. It seems 1.4 allows for a list of SPDX expressions (which seems wrong anyways), whereas, like you said, 1.5 requires a list of licenses or a single expression.

[1.4 Spec](https://cyclonedx.org/docs/1.4/json/#tab-pane_components_items_licenses_items_oneOf_i0)

[1.5 Spec](https://cyclonedx.org/docs/1.5/json/#tab-pane_components_items_licenses_oneOf_i1)

Does this match you're understanding of the specs?

Reply:
Oh yeah. Looks like that is new 1.5 restriction. I remember the rationale for enforcing a single expression. We may have to try to fix this bug in index.js or elsewhere.

Reply:
There are a few situations that I can think of:

1. A component has multiple licenses and a SPDX expression
2. A component has multiple SPDX expressions and multiple licenses
3. A component has multiple licenses, no SPDX expressions
4. A component has one SPDX expression

For 1, it may make sense to ignore the license list and serve the single SPDX expression (the expression takes precedence).

For 2, one would have to choose which expression to persist. It could be simple, but arbitrary, like choosing the first entry, or more complex, like choosing the most permissive or open.

For 3 and 4, those behaviors are known.

Reply:
2) If there are legitimately multiple expressions (with a space or bracket), then we can use the first one and show a warning.